x-pack/plugins/apm/server/lib/helpers/get_ml_setup.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isActivePlatinumLicense } from '../../../common/license_check';
+import { APMRouteHandlerResources } from '../../routes/typings';
+
+export interface MlSetup {
+  mlSystem: any;
+  anomalyDetectors: any;
+  modules: any;
+}
+
+export async function getMlSetup({
+  plugins,
+  context,
+  request,
+}: APMRouteHandlerResources) {
+  const [coreContext, licensingContext] = await Promise.all([
+    context.core,
+    context.licensing,
+  ]);
+
+  return plugins.ml && isActivePlatinumLicense(licensingContext.license)
+    ? {
+        mlSystem: plugins.ml.setup.mlSystemProvider(
+          request,
+          coreContext.savedObjects.client
+        ),
+        anomalyDetectors: plugins.ml.setup.anomalyDetectorsProvider(
+          request,
+          coreContext.savedObjects.client
+        ),
+        modules: plugins.ml.setup.modulesProvider(
+          request,
+          coreContext.savedObjects.client
+        ),
+      }
+    : undefined;
```

Comment:
You can get rid of the indentation like

```suggestion
  if (!plugins.ml || !isActivePlatinumLicense(licensingContext.license)) {
    return;
  }

  return {
    mlSystem: plugins.ml.setup.mlSystemProvider(
      request,
      coreContext.savedObjects.client
    ),
    anomalyDetectors: plugins.ml.setup.anomalyDetectorsProvider(
      request,
      coreContext.savedObjects.client
    ),
    modules: plugins.ml.setup.modulesProvider(
      request,
      coreContext.savedObjects.client
    ),
  };
```

Reply:
Done like this, it complains that plugins.ml might be undefined

Reply:
Ah, you might have to do

```suggestion
  const mlPlugin = plugins.ml;
  if (!mlPlugin || !isActivePlatinumLicense(licensingContext.license)) {
    return;
  }

  return {
    mlSystem: mlPlugin.setup.mlSystemProvider(
      request,
      coreContext.savedObjects.client
    ),
    anomalyDetectors: mlPlugin.setup.anomalyDetectorsProvider(
      request,
      coreContext.savedObjects.client
    ),
    modules: mlPlugin.setup.modulesProvider(
      request,
      coreContext.savedObjects.client
    ),
  };
```